### PR TITLE
Add support for resource links

### LINKED ISSUE #1079: ARM: Add management link commands

### PR DESCRIPTION
**Pull request by brendandburns**
Closes #1079.

### CODE REVIEW

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -509,6 +511,42 @@ def update_lock(name, resource_group_name=None,
     return lock_client.management_locks.create_or_update_at_resource_group_level(
         resource_group_name, name, params)
 
+def create_resource_link(link_id, target_id, notes=None):
+    '''
+    :param target_id: The id of the resource link target.
+    :type target_id: str
+    :param notes: Notes for this link.
+    :type nodtes: str
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: typo: notes

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py [hunk 2]
```diff
@@ -509,6 +511,42 @@ def update_lock(name, resource_group_name=None,
...
+    links_client = _resource_links_client_factory().resource_links
+    properties = ResourceLinkProperties(target_id, notes)
+    links_client.create_or_update(link_id, properties)
+
+def update_resource_link(link_id, target_id=None, notes=None):
+    '''
+    :param target_id: The id of the resource link target.
+    :type target_id: str
+    :param notes: Notes for this link.
+    :type nodtes: str
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: typo: notes

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py (hunk 1 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: done.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py (hunk 2 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: done.

**Comment by brendandburns** [comment]
Comments addresed, please take another look.

Thanks!

**Comment by tjprescott** [comment]
@brendandburns it looks like you need to rebase and resolve conflicts before this can be merged.

**Comment by brendandburns** [comment]
@tjprescott rebased...

**Pull request merged**